src/Button/Button.js

```diff
@@ -25,12 +27,19 @@ const Button = ({
         [`${styles.active} ${classes.active}`]: isActive,
         [styles.bordered]: border,
         [styles.fluid]: fluid,
-        [styles.disabled]: disabled
+        [styles.disabled]: disabled,
+        [styles.loading]: isLoading
       })}
       ref={forwardedRef}
       {...props}
     >
       {children}
+      {isLoading && (
+        <Loader
+          variant={variant === 'fill' ? 'light' : 'dark'}
```

Comment:
Since loader color depends on a `fill` variant, we can specify necessary background (via css variable) of the dot in the `Buttons`'s fill class. 
```css
.fill { 
....
--dot-bg: var(--white);
}
``` 
and then, in dot main class 
```css
.dor {
....
background: var(--dot-bg, var(--mirage));
}
```
By doing this we will remove all conditionals from the main thread